### PR TITLE
Implement `count` for SQLite

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Also, optimize `count` for PostgreSQL.

## Readiness checklist

- [x] I added/updated unit tests.
- [ ] I added/updated integration/compatibility tests.
- [x] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/types/document.go [hunk 1]
```diff
@@ -236,6 +236,7 @@ func (d *Document) Has(key string) bool {
 
 // Get returns a value at the given key.
 // If the key is duplicated, it panics.
+// It returns nil for nil Document.
```
Comment (rumyantseva) [inline internal/types/document.go:239]: It returns `nil` and error, right?

File: internal/handlers/sqlite/msg_count.go [hunk 2]
```diff
@@ -16,12 +16,68 @@ package sqlite
...
+	_, res, err := iter.Next()
+	if errors.Is(err, iterator.ErrIteratorDone) {
+		err = nil
+	}
+
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	count, _ := res.Get("count")
```
Comment (rumyantseva) [inline internal/handlers/sqlite/msg_count.go:71]: is the error was `ErrIteratorDone` could it be possible that `res` is `nil` here?

**Review by rumyantseva — COMMENTED** [review]
LGTM overall, just a couple of comments.

File: internal/types/document.go (hunk 1 above)
Comment (AlekSi) [reply internal/types/document.go:239]: The returned error is described melow

File: internal/handlers/sqlite/msg_count.go (hunk 2 above)
Comment (AlekSi) [reply internal/handlers/sqlite/msg_count.go:71]: Yes

**Review by chilagrow — APPROVED** [review]
Looks good to me!

File: internal/handlers/sqlite/msg_count.go (hunk 2 above)
Comment (chilagrow) [reply internal/handlers/sqlite/msg_count.go:71]: And `count` will be zero.

**Review by rumyantseva — APPROVED** [review]
LGTM

**Pull request merged**